xdsl/passes.py

```diff
@@ -92,7 +92,11 @@ def from_pass_spec(cls: type[ModulePassT], spec: PipelinePassSpec) -> ModulePass
 
         # if not all args were removed we raise an error
         if len(spec.args) != 0:
-            raise ValueError(f'Unrecognised pass arguments "{", ".join(spec.args)}"')
+            arguments_str = ", ".join(f'"{arg}"' for arg in spec.args)
+            fields_str = ", ".join(f'"{field.name}"' for field in fields)
+            raise ValueError(
+                f"Pass arguments [{arguments_str}] not found in [{fields_str}]"
+            )
```

Comment:
This feels a bit snuck in here. Can you open a separate PR? Or atleast test it?

Reply:
Ah sorry this was a part of my debugging while working on this, I'll split it out.

- [ ] split out